aptos-move/framework/aptos-token/sources/token_coin_swap.move

```diff
@@ -64,6 +65,14 @@ module aptos_token::token_coin_swap {
         coin_type_info: TypeInfo,
     }
 
+    public fun is_listing_exist<CoinType>(
```

Comment:
curious why marketplace want to verify this at contract level. This is likely to be a lot of address and token_id to verify. would that be too expensive?

Reply:
We have supported batch buy lots of nft in one transaction, when some assets in demand, it will cause lots of fail txn due to some assets not listing anymore in buying list, and this lead to a terrible experience for customers.
Another solution would be store another listing information in marketplace contract, but I think it's not a good way to duplicate that info on chain.

Reply:
`does_listing_exist`

`is` is not grammatically correct.

Reply:
thx for pointing out that, updated.